fix(docs-infra): prevent search-bar from overlapping nav-items

gkalpak opened the pull request:
As part of angular.io's responsive layout, the menu shown in the top-bar is collapsed into the sidenav on narrow screens at the point where the search-bar (on the right side of the top-bar) would overlap with the menu's nav-items.

Previously, the value used as break-point would work on marketing pages, where the hamburger button is not shown on wide screens. However, on docs pages (where the hamburger button is always shown, pushing the menu further to the right), the search-bar would still overlap the menu nav-items on some resolutions.

This commit fixes it by raising the screen width threshold at a value that ensures there is no overlap even on pages where the hamburger button is visible alongside the top-bar menu.

Fixes #37667.

##
This PR also includes some refactorings in preparation of the fix. See individual commits for details.

petebacondarwin commented on aio/src/app/app.component.spec.ts:
```diff
@@ -186,7 +194,7 @@ describe('AppComponent', () => {
       };
 
       describe('when side-by-side (wide)', () => {
```
```suggestion
      describe('when view is wide', () => {
```

petebacondarwin replied:
there is probably an equivalent `narrow` describe block in this file too.

petebacondarwin commented on aio/src/app/app.component.spec.ts:
```diff
@@ -186,7 +194,7 @@ describe('AppComponent', () => {
 
       describe('when side-by-side (wide)', () => {
-        beforeEach(() => resizeTo(sideBySideBreakPoint + 1));  // side-by-side
+        beforeEach(() => resizeTo(dockSideNavWidth + 1));  // side-by-side
```
```suggestion
        beforeEach(() => resizeTo(dockSideNavWidth + 1));  // wide view
```

petebacondarwin commented on aio/src/app/app.component.spec.ts:
```diff
@@ -233,7 +241,7 @@ describe('AppComponent', () => {
       });
 
       describe('when NOT side-by-side (narrow)', () => {
```
```suggestion
      describe('when view is narrow', () => {
```

petebacondarwin commented on aio/src/app/app.component.spec.ts:
```diff
@@ -233,7 +241,7 @@ describe('AppComponent', () => {
 
       describe('when NOT side-by-side (narrow)', () => {
-        beforeEach(() => resizeTo(sideBySideBreakPoint - 1)); // NOT side-by-side
+        beforeEach(() => resizeTo(dockSideNavWidth - 1)); // NOT side-by-side
```
```suggestion
        beforeEach(() => resizeTo(dockSideNavWidth - 1)); // narrow view
```

petebacondarwin commented on aio/src/app/app.component.spec.ts:
```diff
@@ -292,51 +300,51 @@ describe('AppComponent', () => {
 
```
This describe block could be:

```
describe('when changing from narrow to wide view', () => {
```

mary-poppins commented:
You can preview bfe4764 at https://pr37938-bfe4764.ngbuilds.io/.

petebacondarwin commented on aio/src/styles/1-layouts/_top-menu.scss:
```diff
@@ -1,5 +1,5 @@
 // VARIABLES
-$hamburgerShownMaxWidth: 991px;
+$hamburgerShownMaxWidth: 1047px;
 $hamburgerHiddenMinWidth: $hamburgerShownMaxWidth + 1;
```
Ideally this should have the same name (or as close as possible to the constant in `app.component.ts` (`showTopMenuWidth`) and I would argue that it would be better for the previous line to be computed from this line, rather than the other way around so that the same constant value (i.e. `1048`) is used in both.

The reasoning is that there is no clear indication that these need to be kept in sync. One could imagine searching for the equivalent name or value and not finding anything.

gkalpak replied:
Good point :+1:

petebacondarwin reviewed (approved):
A few minor suggestions... I never liked that `side-by-side` thing so thanks for getting rid of that.

gkalpak committed:
refactor(docs-infra): use Sass variable for top-bar hamburger button show/hide threshold

Use a Sass variable for the screen width break-point at which the
top-bar hamburger button is hidden/shown. This allows more easily
updating the break-point.

gkalpak committed:
refactor(docs-infra): decouple showing the top-menu in top-bar from showing the sidenav

As part of angular.io's responsive layout, the following rules are
applied:
- On wide screens, a menu is shown in the top-bar and the sidenav is
  shown side-by-side with the docs content.
- On narrow screens, the top-menu is moved from the top-bar to the
  sidenav and the sidenav is closed by default and floats over the
  content when manually opened.

Previously, the break-points at which the top-menu was shown in the
top-bar and the sidenav was shown side-by-side with the content were the
same (using a single variable).

This commit decouples the two break-points to make it possible to use
different values in the future.

gkalpak committed:
fix(docs-infra): prevent search-bar from overlapping nav-items

As part of angular.io's responsive layout, the menu shown in the top-bar
is collapsed into the sidenav on narrow screens at the point where the
search-bar (on the right side of the top-bar) would overlap with the
menu's nav-items.

Previously, the value used as break-point would work on marketing pages,
where the hamburger button is not shown on wide screens. However, on
docs pages (where the hamburger button is always shown, pushing the menu
further to the right), the search-bar would still overlap the menu
nav-items on some resolutions.

This commit fixes it by raising the screen width threshold at a value
that ensures there is no overlap even on pages where the hamburger
button is visible alongside the top-bar menu.

Fixes #37937

gkalpak committed:
fixup! refactor(docs-infra): decouple showing the top-menu in top-bar from showing the sidenav

mary-poppins commented:
You can preview 2112df4 at https://pr37938-2112df4.ngbuilds.io/.
